src/config/app.ts

```diff
@@ -0,0 +1,328 @@
+        })
+    });
+
+    return jobs;
+}
+function isJobExcluded(job: Job, exclusions: JobToExcludeFromFile[]) {
+    if (exclusions.length === 0) {
+        return false;
+    }
+
+    return exclusions
+        .filter((exclude) => job.name === exclude.name)
+        .length > 0;
+}
+
+function createJob(name: string, command: string, phpVersion: InstallablePhpVersionType, composerDependency: ComposerDependency, config: Config): Job
+{
+    return {
+        name: name,
+        action: ACTION,
+        operatingSystem: OPERATING_SYSTEM,
+        job: {
+            command: command,
+            php: phpVersion,
+            phpExtensions: config.phpExtensions,
+            phpIni: config.phpIni,
+            composerDependency: composerDependency,
+            ignorePhpPlatformRequirement: config.ignorePhpPlatformRequirements[phpVersion] ?? false,
+            additionalComposerArguments: config.additionalComposerArguments
+        }
+    };
+}
+
+function createJobs(config: Config, tool: Tool): Job[] {
+    const jobs: Job[] = [];
+
+    if (tool.executionType === ToolExecutionType.STATIC) {
+        const lockedOrLatestDependency: ComposerDependency = config.lockedDependenciesExists ? ComposerDependency.LOCKED : ComposerDependency.LATEST;
+        return [createJob(tool.name, tool.command, config.minimumPhpVersion, lockedOrLatestDependency, config)];
+    }
+
+    if (tool.executionType === ToolExecutionType.MATRIX) {
+        if (config.lockedDependenciesExists) {
+            jobs.push(createJob(tool.name, tool.command, config.minimumPhpVersion, ComposerDependency.LOCKED, config));
+        }
+
+        config.versions.forEach(function (version): void {
+            jobs.push(createJob(tool.name, tool.command, version, ComposerDependency.LOWEST, config))
+            jobs.push(createJob(tool.name, tool.command, version, ComposerDependency.LATEST, config))
+        })
+    }
+
+    return jobs;
+}
+
+export function createChecksForKnownTools(config: Config, tools: Tool[]): Job[] {
+    return tools
+        .map((tool) => createJobs(config, tool))
+        .flat(1);
+}
+
+function createNoOpCheck(appConfig: Config): Job {
+    return {
+        name: 'No checks',
+        operatingSystem: OPERATING_SYSTEM,
+        action: ACTION,
+        job: {
+            php: appConfig.stablePhpVersion,
+            phpExtensions: [],
+            command: "",
+            composerDependency: ComposerDependency.LOCKED,
+            phpIni: [],
+            ignorePhpPlatformRequirement: appConfig.ignorePhpPlatformRequirements[appConfig.stablePhpVersion] ?? false,
+            additionalComposerArguments: appConfig.additionalComposerArguments,
+        }
+    };
+}
+
+export function gatherChecks(config: ConfigurationFromFile, appConfig: Config, tools: Tool[]): [Job, ...Job[]] {
+    if (isExplicitChecksConfiguration(config)) {
+        const checks = (config.checks?.map((job) => convertJobFromFileToJobs(job, appConfig)) ?? []).flat(1);
```

Comment:
since these are explicit checks, I don't care about duplicates.
If the project wants to execute the same check multiple times, thats not our problem?